docs/src/main/sphinx/connector/iceberg.rst

```diff
@@ -735,6 +735,22 @@ The output of the query has the following columns:
     - ``array(integer)``
     - The set of field IDs used for equality comparison in equality delete files
 
+``$all_files`` table
+^^^^^^^^^^^^^^^^^^^^
+
+The ``$all_files`` table provides a detailed overview of the data file in all snapshots of the Iceberg table.
+The output of this system table can contain duplicate entries, as corresponding snapshots can refer to the same data files.
+This table is handy to debug the data issues, where the snapshot is rolled back/ bring back to the future from the Trino/Spark
+and now trying to understand what all data-files are present, and is there any implication because of the optimization/rollback operations.
```

Comment:
It would be nice to point to the user a sample Trino SQL query where to employ this table.
Same as in the Iceberg docs: https://iceberg.apache.org/docs/latest/spark-queries/

Reply:
@findinpath typical usage of this the all_files table will be by finding the file_path

```
select file_path from "tbl$all_files"
```

and then use the file_path to query the parquet file, if debugging that data issue.
or use the output if someone wants to automate the cleanup/optimization process for the orphan files.

So, do you think we should add this simple query and example as well?

Reply:
I think that would be good

Reply:
> So let's say I have a table, and I do some insertions. And then I perform some deletes (maybe a whole partition which was the case here). Then I want to see how many files and how many bytes could potentially be removed from storage if we removed table history to the current snapshot.

![image](https://user-images.githubusercontent.com/57260417/157813875-b2437cd7-ac09-4783-99f0-940202548fa3.png)

> So I take relevant information from `$all_data_files` except the relevant information from active `$files`

![image](https://user-images.githubusercontent.com/57260417/157813902-5e9d17e7-8ad0-4aee-9630-8dc660afb411.png)

> We could actually also further join that from the snapshots table and then determine how many snapshots I could retain to reach a certain total disk size on HDFS for example.

Reply:
Let's prepare some well documented SQL queries in the docs to show our users what they can do with this metadata table.

Reply:
@findinpath thanks, this is a very good example!

I have a request, if this example and the preceding one (that I explained - cleanup service one) and maybe a few more we can add in a separate PR, only for documentation.
In that PR we can add examples for other metadata tables like `$file` etc. as well.